File: Sokoban_2.0_Console_Tester/Sokoban_2.0_Console_Tester/Features/Feature.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sokoban_2._0_Console_Tester
{
    public interface Feature
    {
        // Interacting with the Moveable object that Gets on any type of feature
        void Interact(Moveable Moveable);

        void Print(string s, int x, int y);

        string GetFeatureString();
    }
}

File: sokoban/src/hu/bme/sch/sokoban/game/features/Feature.java
package hu.bme.sch.sokoban.game.features;

import hu.bme.sch.sokoban.game.entities.Movable;

public interface Feature {
    // Interacting with the movable object that gets on any type of feature
    void interact(Movable movable);

    void print(String s, int x, int y);

    String getFeatureString();
}